app/api/v0/providers.rb

```
@@ -10,6 +10,22 @@ def oauth_to_json
         params['api_metadatum_attributes']['oauth_info'] = oauth
         params
       end
+
+      def initialize_user
+        user = User.find_or_initialize_by(email: params[:email])
+        create_internal_consumer(user) unless user.persisted?
+        user
+      end
+
+      def create_internal_consumer(user)
+        user.assign_attributes(first_name: params[:firstName], last_name: params[:lastName])
+        user.save
+        user.consumer = Consumer.create(description: 'Internal User', organization: 'Lighthouse')
+      end
+
+      def active_api?(api_name)
+        Api.kept.filter { |a| a.api_metadatum.present? }.map(&:name).uniq.include?(api_name)
+      end
```

should we move this to the API model? or is it relevant here.

---

`Api.kept` is what we want.

---

[b86f0cf](https://github.com/department-of-veterans-affairs/lighthouse-platform-backend/pull/207/commits/b86f0cf67942d61d6946837ec98b5efddda8bdf0) ?

---

close but not quite
that would raise a 500, we probably want a 404 in this instance
should be able to change
```
        api = Api.find_by(name: params[:providerName])
        raise 'Invalid API' unless api.kept?
```
to
```
        api = Api.kept.find_by!(name: params[:providerName])
```
which should raise an activerecord not found exception thats already handled here
https://github.com/department-of-veterans-affairs/lighthouse-platform-backend/blob/API-18749_third_party_endpoints/app/api/base.rb#L15

---

[d6b9893](https://github.com/department-of-veterans-affairs/lighthouse-platform-backend/pull/207/commits/d6b98939148ed9006fff08d4dac17bcc453cf722)